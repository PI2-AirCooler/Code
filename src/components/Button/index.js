
import React from 'react';
import {
  ButtonContainer,
  ButtonText,
} from './style';

function Button({buttonColor, buttonText, onPress}) {
  return (
    <ButtonContainer 
      color={buttonColor}
      onPress={onPress}
    >
      <ButtonText>
        {buttonText}
      </ButtonText>
    </ButtonContainer>
  );
}

export default Button;
