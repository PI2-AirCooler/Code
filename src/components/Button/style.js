import styled from 'styled-components/native';
import normalize from 'react-native-normalize';

export const ButtonContainer = styled.TouchableOpacity`
  width: 70%;
  height: ${normalize(50)}px;
  justify-content: center;
  align-items: center;
  border-radius: ${normalize(7)}px;
  background-color: ${(props) => props.color || "#57B0E2"};
`;

export const ButtonText = styled.Text`
  color: #FFF;
  font-size: ${normalize(20)}px;
  font-weight: bold;
`;
