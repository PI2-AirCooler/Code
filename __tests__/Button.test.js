/**
 * @format
 */

 import 'react-native';
 import React from 'react';
 import Button from '../src/components/Button';
 
 // Note: test renderer must be required after react-native.
 import renderer from 'react-test-renderer';
 
 it('Button renders correctly', () => {
   const tree = renderer
     .create(<Button />)
     .toJSON();
   expect(tree).toMatchSnapshot();
 });
 