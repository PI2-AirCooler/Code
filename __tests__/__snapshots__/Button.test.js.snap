// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#57B0E2",
      "borderRadius": 14,
      "height": 100,
      "justifyContent": "center",
      "opacity": 1,
      "width": "70%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontSize": 40,
          "fontWeight": "bold",
        },
      ]
    }
  />
</View>
`;
